runtime/lua/vim/treesitter.lua

```diff
@@ -438,7 +446,7 @@ end
 ---@param lang (string|nil) Language of the parser (default: from buffer filetype)
 function M.start(bufnr, lang)
   bufnr = bufnr or api.nvim_get_current_buf()
-  local parser = M._get_parser(bufnr, lang)
+  local parser = M.get_parser(bufnr, lang, { error = false })
   if not parser then
     vim.notify('No parser for the given buffer.', vim.log.levels.WARN)
```

Comment:
I think this `vim.notify()` also suffers from the similar issue as `vim.notify_once()` in the `get_parser()`.

Prior to #30313 the call `local has_ts = pcall(vim.treesitter.start, 0)` could be used to start tree-sitter in the buffer while falling back to something else in case of `has_ts == false`. Now it does not fail silently and there is no good way of suppressing the notification. Possible solutions may be:
- Suggest to first call `vim.treesitter.get_parser` to check if there is a parser for the buffer and possibly fall back to something else. This leads to essentially calling `get_parser` twice if there *is a parser, which is not ideal.
- Show no notification if there is no parser (i.e. fail silently).

Not sure what is the best approach here. I'd probably choose a silent fail here.

Reply:
> I think this vim.notify() also suffers from the similar issue as vim.notify_once() in the get_parser().

Yes, but `start` is something that _users_ call (possibly interactively), so they expect feedback -- and `vim.notify` is what we have for this. Let's keep focused on the low-level `get_parser` function.

Reply:
> Let's keep focused on the low-level `get_parser` function.

This was changed in #30313, so I thought it is reasonable to be discussed here.

Reply:
Yes, but correctly so. _That_ is where the warning _should_ be triggered (at top of the stack).

Reply:
But maybe I am wrong, and throwing errors and catching them (and the error message) with `pcall` is really the correct way of such "exception handling"? In this case, we should simply revert.

Reply:
You both bring up great points. Is there a consensus here? I also feel like not throwing an error in `start` is breaking, as with `get_parser`

Reply:
If we want to move away from pcall the I feel we shouldn't be calling vim.notify at all, at least for `start()`.

Reply:
I agree- I mostly meant in terms of erroring, should we still throw in `start`? I hadn't considered that this changes the behavior of `start` as well as `get_parser`

Reply:
There's two issues here:
1. Signaling to the calling function that a parser couldn't be found.
2. Signaling to the _user_ that a parser couldn't be found.

A change in 1. would be considered breaking (code), but a change in 2. would _not_. So no user-facing warning here should be added. If we want to prevent breaking code, we will _have to_ throw an `error`; if we don't, we simply document this and brace for impact.

Reply:
> if we don't, we simply document this and brace for impact.

Sounds good to me? :)

Reply:
Actually, I think it would be best to match the default behavior of `get_parser` (which would error until 0.12) because right now `start` returns nil even upon success. Also keeping backwards compatibility here would remove some headache

Reply:
Yes, being consistent is good. I'll think about how I want to handle this for `language.add()`, which is a lower-level API.